dds/src/implementation/rtps/types.rs

```diff
@@ -139,47 +139,154 @@ impl From<[u8; 4]> for EntityId {
-pub const BUILT_IN_PARTICIPANT: EntityKind = 0xc1;
-pub const USER_DEFINED_WRITER_WITH_KEY: EntityKind = 0x02;
-pub const BUILT_IN_WRITER_WITH_KEY: EntityKind = 0xc2;
-pub const USER_DEFINED_WRITER_NO_KEY: EntityKind = 0x03;
-#[allow(dead_code)]
-pub const BUILT_IN_WRITER_NO_KEY: EntityKind = 0xc3;
-#[allow(dead_code)]
-pub const USER_DEFINED_READER_WITH_KEY: EntityKind = 0x07;
-pub const BUILT_IN_READER_WITH_KEY: EntityKind = 0xc7;
-#[allow(dead_code)]
-pub const USER_DEFINED_READER_NO_KEY: EntityKind = 0x04;
-#[allow(dead_code)]
-pub const BUILT_IN_READER_NO_KEY: EntityKind = 0xc4;
-pub const USER_DEFINED_WRITER_GROUP: EntityKind = 0x08;
-pub const BUILT_IN_WRITER_GROUP: EntityKind = 0xc8;
-pub const USER_DEFINED_READER_GROUP: EntityKind = 0x09;
-pub const BUILT_IN_READER_GROUP: EntityKind = 0xc9;
-// Added in comparison to the RTPS standard
-pub const BUILT_IN_TOPIC: EntityKind = 0xca;
-pub const USER_DEFINED_TOPIC: EntityKind = 0x0a;
+impl serde::Serialize for EntityKind {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serde::Serialize::serialize(
+            &Into::<u8>::into(*self),
+            serializer,
+        )
+    }
+}
+
+struct EntityKindVisitor;
+
+impl<'de> serde::de::Visitor<'de> for EntityKindVisitor {
+    type Value = EntityKind;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_str(&format!("value must be valid EntityKind"))
+    }
+
+    fn visit_u8<E>(self, value: u8) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        Ok(value.into())
+    }
+}
+
+impl<'de> serde::Deserialize<'de> for EntityKind {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where D: serde::Deserializer<'de>
+    {
+        deserializer.deserialize_u8(EntityKindVisitor)
+    }
+}
+
+impl From<EntityKind> for u8 {
+    fn from(value: EntityKind) -> Self {
+        match value {
+            EntityKind::UserDefinedUnknown => 0x00,
+            EntityKind::BuiltInUnknown => 0xc0,
+            EntityKind::BuiltInParticipant => 0xc1,
+            EntityKind::UserDefinedWriterWithKey => 0x02,
+            EntityKind::BuiltInWriterWithKey => 0xc2,
+            EntityKind::UserDefinedWriterNoKey => 0x03,
+            EntityKind::BuiltInWriterNoKey => 0xc3,
+            EntityKind::UserDefinedReaderWithKey => 0x07,
+            EntityKind::BuiltInReaderWithKey => 0xc7,
+            EntityKind::UserDefinedReaderNoKey => 0x04,
+            EntityKind::BuiltInReaderNoKey => 0xc4,
+            EntityKind::UserDefinedWriterGroup => 0x08,
+            EntityKind::BuiltInWriterGroup => 0xc8,
+            EntityKind::UserDefinedReaderGroup => 0x09,
+            EntityKind::BuiltInReaderGroup => 0xc9,
+            EntityKind::UserDefinedTopic => 0x0a,
+            EntityKind::BuiltInTopic => 0xca,
+        }
+    }
+}
+
+impl From<u8> for EntityKind {

```

Comment:
This should be a TryFrom trait

Reply:
Yes. That's what need to come in another PR